jwst/model_blender/blendmeta.py

```diff
@@ -3,328 +3,16 @@
-
-    # apply rules to datamodels metadata
-    new_meta, newtab = final_rules.apply(input_models)
-
-    if len(newtab) > 0:
-        # Now merge the results for all the tables into single table extension
-        new_table = fits.BinTableHDU.from_columns(newtab)
-        new_table.header['EXTNAME'] = 'HDRTAB'
-    else:
-        new_table = None
-
-    if close_models:
-        for model in input_models:
-            model.close()
-
-    return new_meta, new_table
-
-
-def cat_headers(hdr1, hdr2):
-    """
-    Create new `astropy.io.fits.Header` object from
-    concatenating 2 input Headers
-    """
-    nhdr = hdr1.copy()
-    for c in hdr2.cards:
-        nhdr.append(c)
-
-    return fits.Header(nhdr)
-
-
-def extract_filenames_from_product(product):
-    """
-    Returns the list of filenames with extensions of input observations that
-    were used to generate the product.
-    """
-    asn_table = product.meta.asn.table_name
-    asn = associations.load_asn(asn_table)
-    prod = asn['products'][0]
-    fnames = [m['expname'] for m in prod['members']
-              if m['exptype'].upper() == 'SCIENCE']
-    return fnames
-
-
-def build_tab_schema(new_table):
-    """Return new schema definition that describes the input table."""
-    hdrtab = OrderedDict()
-    hdrtab['title'] = 'Combined header table'
-    hdrtab['fits_hdu'] = 'HDRTAB'
-    datatype = []
-    for col in new_table.columns:
-        cname = col.name
-        ctype = convert_dtype(str(col.dtype))
-        c = OrderedDict()
-        c['name'] = cname
-        c['datatype'] = ctype
-        datatype.append(c)
-    hdrtab['datatype'] = datatype
-
-    return hdrtab
-
-
-def convert_dtype(value):
-    """Convert numarray column dtype into YAML-compatible format description"""
-    if 'S' in value:
-        # working with a string description
-        str_len = int(value[value.find('S') + 1:])
-        new_dtype = ['ascii', str_len]
-    else:
-        new_dtype = str(value)
-
-    return new_dtype
-
-
-def _build_schema_ignore_list(schema):
-    """ Create a list of metadata that should be ignored when blending.
-
-    Parameters
-    ----------
-    schema : JSON schema fragment
-        The schema in which to search.
-
-    Returns
-    -------
-    results : list
-        List with schema attributes that needs to be ignored
-
-    """
-    def build_rules_list(subschema, path, combiner, ctx, recurse):
-        # Only interpret elements of the meta component of the model
-        if len(path) > 1 and path[0] == 'meta' and 'items' not in path:
-            attr = '.'.join(path)
-            if subschema.get('properties'):
-                return  # Ignore ObjectNodes
-            kwtype = subschema.get('type')
-            if kwtype == 'array':
-                results.append(attr)
-        else:
-            return
-
-    results = []
-    dm_schema.walk_schema(schema, build_rules_list, results)
-    return results
+def blendmodels(product, inputs, ignore=None):
```

Comment:
Since this is now only a single function of five lines, would it perhaps be easier and enhance readability to delete this file and move the function elsewhere, e.g. into blender.py?

Reply:
Actually, is this used anywhere besides the test suite?  Could it be removed entirely?

Reply:
Thanks! This function is used in cube_build/ifu_cube.py. See:https://github.com/spacetelescope/jwst/pull/8759/files#diff-4eb3446a610f71b161f1fdbe2ff28934ea50381e54a060851e440882bb272ae1R2366

Reply:
huh, not sure why that didn't pop up when I clicked the button that's supposed to show me where it's used. thanks.